### PR TITLE
Ammending meson.build files to fix linking issue - issue38

### PR DESCRIPTION
**Pull request by russnic**
*Issue #, if available:*
- See: https://github.com/amzn/amazon-s3-gst-plugin/issues/38

*Description of changes:*
- Updating meson.build files to fix link issues

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by louisharris — APPROVED** [review]

File: src/meson.build [hunk 1]
```diff
@@ -15,7 +15,7 @@ symbol_export_define = ['-DGST_EXPORTS']
 
 credentials = library('gstawscredentials-@0@'.format(apiversion),
   ['gstawscredentials.cpp'],
-  dependencies : [aws_cpp_sdk_sts_dep, gst_dep],
+  dependencies : [aws_cpp_sdk_sts_dep, gst_dep, aws_c_common_dep, aws_crt_cpp_dep],
```
Comment (loganek) [inline src/meson.build:18]: just out of curiosity; why is that needed? isn't it an implicit dependency of `aws-cpp-sdk-sts`?

Comment (russnic) [reply src/meson.build:18]: I’ve spoken with the engineer who requested this, and they advised it's because they're building with yocto for embedded. They tried to rebuild without the dependencies, and it still fails. For context this is where they have their [patch](https://github.com/aws4embeddedlinux/meta-aws/blob/master-next/recipes-multimedia/amazon-s3-gst-plugin/files/fix-linking-of-aws-crt-cpp.patch) in their repo.

Comment (russnic) [reply src/meson.build:18]: We have an open issue with the AWS SDK to investigate this see https://github.com/aws/aws-sdk-cpp/issues/2643

Comment (loganek) [reply src/meson.build:18]: Thanks!

**Review by loganek — APPROVED** [review]
LGTM
